### PR TITLE
[node-forge] add types for missing modules; fix incorrect types

### PR DESCRIPTION
**Pull request by timhwang21**
Add typedefs for:
* `jsbn` (bare minimum); API isn't public so should be fine
* `random`

Improve typedefs for:
* `pki` public and private keys
  * Add types for `pki.rsa` keys

Add stubs for: (I didn't fix as these types aren't currently relevant to
my work)
* `pss`
* `pki.ed25519`

Also fixed some incorrect types, e.g. `createBuffer` was mistyped.

Added test case for RSA signing. Also fixed some incorrect tests, and
commented out ones that seemed entirely broken. The two removed lines
don't trigger type errors on master, but if you actually run the file
they will error: `ts-node types/node-forge/node-forge-tests.ts`

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.) **(ran with `ts-node` as trying to run `npm test` seemed to cause a Node memory leak which crashed my machine)**
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/digitalbazaar/forge
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node-forge/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for node-forge 0.7.6
+// Type definitions for node-forge 0.7.7
```
Comment (joeflateau) [inline types/node-forge/index.d.ts]:
```suggestion
// Type definitions for node-forge 0.7.6
```

Comment (joeflateau) [reply types/node-forge/index.d.ts]: https://www.npmjs.com/package/node-forge current v is 0.7.6

**Review by joeflateau — CHANGES_REQUESTED** [review]

File: types/node-forge/index.d.ts (hunk 1 above)
Comment (timhwang21) [reply types/node-forge/index.d.ts]: Oops I wasn't thinking, for some reason I thought types were versions independently. Fixed.

Comment (joeflateau) [reply types/node-forge/index.d.ts]: yeah, i did the same thing 🙃

**Review by joeflateau — APPROVED** [review]

**Pull request merged**